netlify.toml

```diff
@@ -0,0 +1,10 @@
+# Global settings applied to the whole site.
+[build]
+  base    = "/"
+  publish = "app/"
+  command = "npm install && npm run setup"
+
+# Deploy Preview context: All Deploy Previews
+# will inherit these settings.
+[context.set-api.environment]
+  APIHOST = "https://104.199.16.253/api/"
```

Comment:
I think you don't need that. the [context.set-api.environment] will only be used in the set-api branch (that is a specific branch). Even if you merge that back to master, that api won't be used at all in master.
In your example the api in branch-deploy will be used in all branches, something you don't want right?
Am I missing something?